lib/elixir/lib/uri.ex

```diff
@@ -279,10 +280,10 @@ defmodule URI do
       "a str%69ng"
 
   """
-  @spec encode(binary, (byte -> boolean)) :: binary
-  def encode(string, predicate \\ &char_unescaped?/1)
-      when is_binary(string) and is_function(predicate, 1) do
-    for <<char <- string>>, into: "", do: percent(char, predicate)
+  @spec encode(binary, as_boolean(char)) :: binary
```

Comment:
Wait, doesn't this take a fun? 🤔

Reply:
yes. I made the same mistake before. thank you for noting it.